src/Compilers/Core/Portable/Syntax/GreenNode.cs

```diff
@@ -516,7 +516,7 @@ public bool HasAnnotations(IEnumerable<string> annotationKinds)
             return false;
         }
 
-        public bool HasAnnotation(SyntaxAnnotation annotation)
+        public bool HasAnnotation([NotNullWhen(true)] SyntaxAnnotation? annotation)
```

Comment:
If `null` is an expected value here it makes me question if we should have a fast return path for `null` @333fred

Reply:
I found code that assumes this method handles null arguments (by returning `false`), but couldn't find an indication that this was expected to be a common or fast path.

Reply:
Yeah, I'd prefer fast-path changes like that to be driving by traces.